### PR TITLE
15711 n-in-1-out

### PR DESCRIPTION
**Pull request by iviar**
### Requirements

* Filling out this template is required. Pull Requests without a clear description may be closed at the maintainers' discretion.

### Description

handle SINGLENOZZLE with one thermal control from the LCD menu

### Benefits

correct handling of the LCD the number of thermal control (fans) when SINGLENOZZLE is defined (which is the case with one fan, I don't think there is a case of SINGLENOZZLE and multiple fan as the code seem expected rising just a warning and possibly an UI misbehaviour as an array (read) access out of bound happen uncontrolled)

### Related Issues

fix issue #15711 [BUG] when SINGLENOZZLE has one fan (always!?) the LCD menu assumes there are more fans

### CODE REVIEW

**Comment by thinkyhead** [comment]
The original code implies that singlenozzle has "virtual fans" (since the OR condition makes `HAS_FAN1` irrelevant) but I don't see anything that looks like that.

File: Marlin/src/lcd/menu/menu_temperature.cpp [hunk 1]
```diff
@@ -242,14 +242,14 @@ void menu_temperature() {
         EDIT_ITEM_FAST(percent, MSG_EXTRA_FAN_SPEED_1, &thermalManager.new_fan_speed[0], 3, 255);
       #endif
     #endif
-    #if HAS_FAN1 || (ENABLED(SINGLENOZZLE) && EXTRUDERS > 1)
+    #if HAS_FAN1 || (DISABLED(SINGLENOZZLE) && EXTRUDERS > 1)
```
Comment (thinkyhead) [inline Marlin/src/lcd/menu/menu_temperature.cpp]: This condition doesn't get rid of the issue that `HAS_FAN1` is made irrelevant. I will look into the back-history of this code to see why it is shaped the way it is.

**Comment by thinkyhead** [comment]
Ok, so the original conditions were added by @InsanityAutomation in #12853 to couple multiple fan settings with `SINGLENOZZLE` though I am not sure what the benefit of this correlation is. Prior to that change Marlin inferred the existence of a fan —and the option to set its speed— based on having pins for it on the board, plus something to assign it to.

The thing is, with or without single-nozzle, you could have two or more independent fans blowing on opposite sides of the nozzle, for example, though usually this is not done.

Let's wait for some clarification about these conditions from @InsanityAutomation and then decide how to patch it.

**Comment by InsanityAutomation** [comment]
I havnt been over all the code in awhile to be 100% sure nothing has changed, however at the time of writing single nozzle applies the standby temp and fan speed at tool change. For example, when T1 is active, Fan 2 speed is used to control physical fan 1. Hotend 2 temperature is applied to the physical hotend 1. When T0 is active Fan 1 speed and Hotend 1 are used.

The intent is for drastically different materials to have independent fan settings without continually forcing the slicer to relay them on every tool change. If running say ABS which requires nearly no fan around say a TPU piece this could become quite critical.

File: Marlin/src/lcd/menu/menu_temperature.cpp (hunk 1 above)
Comment (iviar) [reply Marlin/src/lcd/menu/menu_temperature.cpp]: the issue is #15711 

I don't understand "HAS_FAN1 is made irrelevant",
it is defined in src/inc/Conditionals_post.h as follow:
#define HAS_FAN1 (PIN_EXISTS(FAN1) && CONTROLLER_FAN_PIN != FAN1_PIN && E0_AUTO_FAN_PIN != FAN1_PIN && E1_AUTO_FAN_PIN != FAN1_PIN && E2_AUTO_FAN_PIN != FAN1_PIN && E3_AUTO_FAN_PIN != FAN1_PIN && E4_AUTO_FAN_PIN != FAN1_PIN && E5_AUTO_FAN_PIN != FAN1_PIN)

while SINGLENOZZLE and EXTRUDERS are user definable in Configuration.h

**Pull request closed without merging**